### PR TITLE
Add option to include defined-by annotation in imports

### LINKED ISSUE #796: Add rdfs:isDefinedBy to imports
ROBOT now has this cool option with merge to add provenance:

```
robot merge -I http://purl.obolibrary.org/obo/pato.owl -I http://purl.obolibrary.org/obo/ro.owl --annotate-defined-by true -o defined-by.owl
```

We should consider adding this by default to ODK.

### PR DESCRIPTION
**Pull request by anitacaron**
Fixes #796

There're two cases:

1. When `use_base_merging: true`, the option is added when merging all mirrors, so then we can get properly where the terms come from.
2. When `use_base_merging` not defined, the option is added during the release process.

### CODE REVIEW

**Comment by anitacaron** [comment]
Can you clarify, @matentzn, please?

**Comment by matentzn** [comment]
This should be configurable at least on import_group level, because the file size will blow up massively with that feature, and we want to at least give some hope that it can be switched off for people for which size matters (github file size limits etc).

Also,  what about the case that you are not using base merging, but want to use this feature when creating the full.owl release? Isnt this another case where we may want this (we could use the same flag from import_group.annotate_defined_by)?

File: template/src/ontology/Makefile.jinja2 [hunk 1]
```diff
@@ -915,7 +915,7 @@ $(ONT).owl: $(ONT)-{{ project.primary_release }}.owl
 {% endif -%}
 
 # ROBOT pipeline that merges imports, including components.
-ROBOT_RELEASE_IMPORT_MODE={% if project.use_edit_file_imports -%}$(ROBOT) merge --input $< {% else -%}
+ROBOT_RELEASE_IMPORT_MODE={% if project.use_edit_file_imports -%}$(ROBOT) merge --input $< --annotate-defined-by true {% else -%}
```
Comment (anitacaron) [inline template/src/ontology/Makefile.jinja2:918]: @matentzn 
> Also, what about the case that you are not using base merging, but want to use this feature when creating the full.owl release? Isnt this another case where we may want this (we could use the same flag from import_group.annotate_defined_by)?

I did this in the variable `ROBOT_RELEASE_IMPORT_MODE`.

**Comment by anitacaron** [comment]
Ok, I'll add the configuration for this.

**Comment by anitacaron** [comment]
@matentzn is ready for review again

File: template/src/ontology/Makefile.jinja2 [hunk 2]
```diff
@@ -915,7 +915,7 @@ $(ONT).owl: $(ONT)-{{ project.primary_release }}.owl
 {% endif -%}
 
 # ROBOT pipeline that merges imports, including components.
-ROBOT_RELEASE_IMPORT_MODE={% if project.use_edit_file_imports -%}$(ROBOT) merge --input $< {% else -%}
+ROBOT_RELEASE_IMPORT_MODE={% if project.use_edit_file_imports -%}$(ROBOT) merge --input $< {% if project.import_group is defined and project.import_group.use_base_merging is false and project.import_group.annotate_defined_by %}--annotate-defined-by true{% endif -%}{% else -%}
```
Comment (matentzn) [inline template/src/ontology/Makefile.jinja2:918]: Is this conditional needed? `and project.import_group.use_base_merging is false`? seems redundant. If you try to annotate an already annotated class, nothing happens!

Comment (matentzn) [reply template/src/ontology/Makefile.jinja2:918]: (but please test this if true) :D

Comment (matentzn) [reply template/src/ontology/Makefile.jinja2:918]: Ok! Looks good then

**Comment by matentzn** [comment]
From now on for changes that affect the pipeline, we should have two independent reviews. To minimise mistakes..

**Review by gouttegd — APPROVED** [review]

**Pull request merged**